apps/pollbook/backend/src/app.ts

```diff
@@ -266,13 +267,36 @@ function buildApi({ context, logger, barcodeScannerClient }: BuildAppParams) {
     async checkInVoter(input: {
       voterId: string;
       identificationMethod: VoterIdentificationMethod;
+      ballotParty: PartyAbbreviation;
```

Comment:
The behavior as written is that all check-ins require a party, even generals, but we only validate and display those for primaries.

This is a little off semantically - what does it mean to have a party selection in a general? But it was a little easier to require this across the board so that we get type safety and don't have to throw existence checks in a bunch of places. I'm still on the fence and may revisit this later to allow `ballotParty` IFF the election is a primary.